identity-service/src/notifications/index.js

```diff
@@ -56,11 +56,14 @@ class NotificationProcessor {
     this.mg = expressApp.get('mailgun')
 
     // Index all blockchain ids
-    await updateBlockchainIds()
+    this.idUpdateTask = updateBlockchainIds()
 
     // Notification processing job
     // Indexes network notifications
     this.notifQueue.process(async (job, done) => {
+      // Await blockchain IDs before indexing notifs
+      await this.idUpdateTask
```

Comment:
would this run every time process is called? that might actually be what we want

Reply:
no it is not re-run, the original promise is re-awaited so it is a no-op after the first completion. We can maybe schedule it to run intermittently? like if it has been x minutes since the last run or something?